views/event/index.ejs

```diff
@@ -3,49 +3,65 @@
 
 <head>
     <% include ../template/head.ejs %>
-        <script src="/js/SearchScript.js"></script>
-        <script src="/js/SortScript.js"></script>
-        <script src="/js/slideshow.js"></script>
-        <link rel="stylesheet" type="text/css" href="/css/slideshow.css" />
+    <link rel="stylesheet" type="text/css" href="/css/slideshow.css"/>
 </head>
 
 <body>
+    <% include ../template/nav.ejs %>
     <div class="container body-layout">
         <div class="row center">
             <h2>Events</h2>
         </div>
-        <div class="slideshow-container">
-        	<% if(eventsSet.length == 0){ %>
-            <p class="error-msg">Sorry! Currently, there are no project available</p>
-        <% } %>
-        <% eventsSet.forEach(function(row){ %>
-        	<!-- Full-width images with number and caption text -->
-			<div class="mySlides fade">
-                <div class="numbertext"><%= row.number %> / <%= eventsSet.length%></div>
-                <div class="container">
-                	<!-- Not quite sure how to include <%= %> in "" -->
-                    <img src="<%= row.img%>" class="slides">
-                    <div class="overlay">
+
+        <% if (eventsSet.length == 0){  %>
+            <p class="error-msg">Sorry! Currently, there are no events available</p>
+        <% } else { %>
+            <!-- Slideshow container -->
+            <div class="slideshow-container">
+
+                <% if (auth([all_user_role.Developer,
+                    all_user_role.ProjectManager,
+                    all_user_role.Admin,
+                    all_user_role.EventExecutive,
+                    all_user_role.ProjectLeader], user_role)) { %>
+                    <div class="col-sm-8-offset" align="right">
+                        <a class="btn primary-button" href="/event/create">Create Event</a>
+                    </div>
+                <% } %>
+
+                <% var counter = 0; %>
+                <% eventsSet.forEach(function(row) { %>
+                    <% counter += 1; %>
+                    <!-- Full-width images with number and caption text -->
+                    <div class="mySlides">
+                        <div class="numbertext"> <%= counter %> / <%= eventsSet.length %> </div>
+                        <% var link = row.link%>
+                        <!-- Add https:// if link does not have it-->
+                        <% if (link.substring(0, 8) != "https://") { link = "https://" + link } %>
```

Comment:
Could you create a new function that does this `url_with_https func(old_url)` and move the validate url and add https to a new js file for future reuse? Thanks

Reply:
add to views/template/head.ejs, where another function `auth = function(auth_roles, user_role)` resides.